pkg/client/session.go

```diff
@@ -36,9 +38,7 @@ func (c *immuClient) OpenSession(ctx context.Context, user []byte, pass []byte,
 		return err
 	}
 
-	if c.ServiceClient == nil {
-		c.ServiceClient = schema.NewImmuServiceClient(c.clientConn)
-	}
+	c.ServiceClient = schema.NewImmuServiceClient(c.clientConn)
```

Comment:
Why this change is needed? There's no way to set custom ServiceClient that way

Reply:
hm, the client API should be revised... while the ServiceClient could be externally set, a new client connection was stablished anyways, and assumed to be there in other parts of the code i.e. when closing the session